design.md

```diff
@@ -1,13 +1,79 @@
 # Design
 
+A grammar for our language:
+```
+<program>    := <state> <program>
+              | <state>
+
+<state>      := state <state_name> <rules>
+
+<rules>      := <rule> <rules>
+              | <rule>
+
+<rule>       := "|" move <dir>
+              | "|" face <dir>
+              | "|" <state_name>
+              | "|" move <dir>, face <dir>
+              | "|" move <dir>, <state_name>
+              | "|" face <dir>, <state_name>
+              | "|" move <dir>, face <dir> <state_name>
+
+<dir>        := north | south | east | west | left | right | forward | backward
+
+<state_name> := [a-zA-Z0-9_]+
+```
 ## Who is the target for this design, e.g., are you assuming any knowledge on the part of the language users?
 
+The target of this design is similar to the target for the original Picobot.
+The language is meant for students with little CS knowledge, since it doesn't
+assume familiarity with CS concepts other than states and state transitions, in
+the same way Picobot does.
+
 ## Why did you choose this design, i.e., why did you think it would be a good idea for users to express the maze-searching computation using this syntax?
 
+We chose this design while thinking about the solution for the maze.  Many of
```

Comment:
What were some of your other ideas like out of curiosity?

Although there is a danger in only thinking about the solution versus extensibility when designing a langugae, I think ultimately the syntax you came up with is expressive enough that it can be used for most programs that would be encoded with the original syntax.